src/Symfony/Component/HttpFoundation/Tests/RequestTest.php

```diff
@@ -326,6 +326,20 @@ public function testGetMimeTypeFromFormat($format, $mimeTypes)
         }
     }
 
+    /**
+     * @dataProvider getFormatToMimeTypeMapProviderForAccept
+     */
+    public function testGetMimeTypeFromFormatWithAccept($format, $mimeTypes, $accepts)
+    {
+        if (null !== $format) {
+            $request = new Request();
+            $request->headers->set('Accept', $accepts);
+            $request->setFormat($format, $mimeTypes);
+            //Only when accept value matches a value verbatim inside of the mimetype array.
+            $this->assertEquals($accepts, $request->getMimeType($format));
```

Comment:
Your test behaves the same against the current code base as it does against my changes....I don't understand what you mean that my changes are not BC.

It passes in both instances because it doesn't care that header is set. That method has never cared if header is set. I'm making it care ONLY IF it's within the $format array and ONLY IF it's present in the values.

Currently it does not exclusively return the value requested by the accept header.

I still don't see the BC break :(

But I do appreciate the concern, as I would also have that concern.